yamlpatch/patch.go

```diff
@@ -232,6 +232,7 @@ func valueToYAMLNode(value interface{}) (*yaml.Node, error) {
 		return nil, err
 	}
 	clearYAMLStyle(node)
+	node.HeadComment = comment
```

Comment:
Will this overwrite an existing comment every time? Should this be conditioned on whether the `comment == ""`? If so, do we also need a way to _remove_ an existing comment?

Reply:
This node is constructed by marshaling the value (which is `interface{}`) into yaml, and then unmarshalling that into a `yaml.Node` ... so prior to this PR, the `node.HeadComment` field would always be empty (no change in behavior for operations that do not define a comment). 

When a comment is included it gets inserted above the value and won't impact other comments in the original document that are not in the scope of that node.

Example of behavior doing a `Replace` on `foo/bar` path with a comment in the operation:
```
foo:
  # some other comment
  bar: # inline comment
  # original comment
  - 10
  - 20
```

```
foo:
  # some other comment
  bar: # inline comment
  # patched comment
  - 10
  - 20
```
vs prior to this PR or without a comment (equivalent):
```
foo:
  # some other comment
  bar: # inline comment
  - 10
  - 20
```